components/board/genericlinux/board.go

```diff
@@ -321,9 +321,9 @@ func (b *Board) reconfigureInterrupts(newConf *LinuxBoardConfig) error {
 			oldInterrupt = nil
 		}
 
-		gpioMapping, ok := b.gpioMappings[config.Name]
+		gpioMapping, ok := b.gpioMappings[config.Pin]
```

Comment:
This is a pitfall of having too many things of type `string` (sometimes called [stringly typed code](https://blog.codinghorror.com/new-programming-jargon/)). If we had `PinName` and `InterruptName` as distinct types that both acted like strings but could not be used interchangeably, this gets  caught by the compiler. Is it worth doing that? Maybe...

Reply:
NICE!!!!

Reply:
(I don't know the answer to this question) but is there a pattern/distinguisher for the strings associated with config.Name vs config.Pin? Maybe you could create a helper function that checks if the strings are of the correct type that way, rather than creating a new type / struct.